src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildACppCliProject.cs

```diff
@@ -166,5 +191,79 @@ private void ChangeTargetFramework(string projectPath, XDocument project, string
                                             .Single().Value = targetFramework;
             }
         }
+
+        private void ConfigureProject(string projectPath, XDocument project, string package, string version, string targetFramework, string[] properties)
+        {
+            ChangeTargetFramework(projectPath, project, targetFramework);
+            AddPackageReference(projectPath, project, package, version);
+            RecordProperties(projectPath, project, properties);
+        }
+
+        private void AddPackageReference(string projectPath, XDocument project, string package, string version)
+        {
+            if (Path.GetExtension(projectPath) == ".vcxproj")
+            {
+                XNamespace ns = project.Root.Name.Namespace;
+                XElement itemGroup = project.Root.Descendants(ns + "ItemGroup").First();
+                itemGroup.Add(new XElement(ns + "PackageReference", new XAttribute("Include", package),
+                                                    new XAttribute("Version", version)));
+
+            }
+        }
+        private void RecordProperties(string projectPath, XDocument project, string[] properties)
+        {
+            if (Path.GetExtension(projectPath) == ".vcxproj")
+            {
+                List<XElement> propertiesElements = new List<XElement>();
+                XNamespace ns = project.Root.Name.Namespace;
+                foreach (var propertyName in properties)
+                {
+                    propertiesElements.Add( new XElement(ns + "LinesToWrite", new XAttribute("Include", @$"{ propertyName }: $({ propertyName })")));
+                }
+
+//                string target = $@"<Target Name='WritePropertyValues' BeforeTargets='AfterBuild'>
+//    <ItemGroup>
+//{propertiesElements}
+//    </ItemGroup>
+//    <WriteLinesToFile
+//      File='$(BaseIntermediateOutputPath)\$(Configuration)\$(TargetFramework)\PropertyValues.txt'
+//      Lines='@(LinesToWrite)'
+//      Overwrite='true'
+//      Encoding='Unicode'
+//      />
+//  </Target>";
```

Comment:
It might be easier to read and less redundant if you did an `XElement.Parse` on a string literal with the whole target instead of building it up in code.

Reply:
The problem I found is that XElement.parse ends up creating an empty namespace (xmlns="") which then breaks the build. I've tried a few different options but I end up setting the namespace on the top level target but still have the empty namespace on the sub nodes.  I found some examples for removing the namespace from every entry but not one for setting it.  Any ideas?

Reply:
ok, found a way to do clean up the namespaces and cleaned up the code to do it in the more readable way.